halo2_proofs/src/poly/evaluator.rs

```diff
@@ -196,6 +201,17 @@ impl<E, F: Field, B: Basis> Evaluator<E, F, B> {
             leaves: &'a HashMap<AstLeaf<E, B>, &'a [F]>,
         }
 
+        impl<'a, E, F: FieldExt, B: Basis> AstContext<'a, E, F, B> {
+            /// Returns the actual size of the chunk we're operating on in this
+            /// context, which may be smaller than `chunk_size`.
+            fn local_chunk_size(&self) -> usize {
+                cmp::min(
+                    self.chunk_size,
+                    self.poly_len - self.chunk_size * self.chunk_index,
+                )
+            }
+        }
+
```

Comment:
After my other suggestion, this becomes unused. Let's remove it from this PR, but keep it in mind as a nice refactor for a future PR (we could simplify the basis trait APIs by refactoring `AstContext` to have a sub-struct storing just the fields necessary for this calculation, which is currently repeated in several places).